### PR TITLE
Factor out a withMigrate wrapper

### PR DESCRIPTION
**Pull request by timleslie**
This PR adds a wrapper function `withMigration` to capture the common pattern used to setup/teardown prisma `Migrate` objects.

### CODE REVIEW

File: packages/adapter-prisma/src/migrations.ts [hunk 1]
```diff
@@ -22,21 +22,34 @@ function runMigrateWithDbUrl<T>(dbUrl: string, cb: () => T): T {
...
+async function withMigration<T>(
+  dbUrl: string,
+  schemaPath: string,
+  cb: (migrate: Migrate) => Promise<T>
+) {
+  await ensureDatabaseExists(dbUrl, path.dirname(schemaPath));
+  const migrate = new Migrate(schemaPath);
+
+  try {
+    return cb(migrate);
```
Comment (mitchellhamilton) [inline packages/adapter-prisma/src/migrations.ts:34]: You need to await the return value here otherwise the stop() is happening at the wrong time

Comment (timleslie) [reply packages/adapter-prisma/src/migrations.ts:34]: Nice catch, thanks!

File: packages/adapter-prisma/src/migrations.ts [hunk 2]
```diff
@@ -22,21 +22,34 @@ function runMigrateWithDbUrl<T>(dbUrl: string, cb: () => T): T {
   }
 }
 
+async function withMigration<T>(
```
Comment (mitchellhamilton) [inline packages/adapter-prisma/src/migrations.ts:25]: Can you rename this to withMigrate?

Comment (timleslie) [reply packages/adapter-prisma/src/migrations.ts:25]: Sure :-)

**Review by mitchellhamilton — APPROVED** [review]

**Pull request merged**